### ISSUE TITLE
Exception when running example code

### ISSUE THREAD

**Issue by stoopdapoop** [wesleythorsen1/dotnet-craigslist#12]
So I've copied the example code from the readme:

```
            var client = new CraigslistStreamingClient();

            var request = new SearchForSaleRequest(
                "losangeles",
                SearchForSaleRequest.Categories.Electronics)
            {
                SearchText = "graphics card",
            };

            await foreach (var posting in client.StreamPostings(request))
            {
                Console.WriteLine($"{posting.Price} : {posting.Title} - {posting.PostingUrl}");
            }
```

But I get an exception after it iterates over a few of the results.

```
Exception thrown: 'System.ArgumentNullException' in System.Private.CoreLib.dll
Exception thrown: 'System.ArgumentNullException' in System.Private.CoreLib.dll
An exception of type 'System.ArgumentNullException' occurred in System.Private.CoreLib.dll but was not handled in user code
Value cannot be null.
```

This is in dotNet 6.0. I can provide more info if needed

**Comment by wesleythorsen1** [comment]
I am unable to reproduce. Do you have more details on the exception (stack trace) or what exact search parameters you used? It's possible that some of the values the `PageParser` is expecting from the craigslist html aren't returned, and is throwing this exception when parsing into a `Posting` object.

**Comment by stoopdapoop** [comment]
Yeah, the above code is the exact code used, I only added a "using DotnetCraigslist;" at the top so it'd compile.

I got version 0.2.0 from nuget in visual studio.

Here's the stack trace:

<HTML>
<head>
<title>Document</title></head>
<body>



  | Name | Value | Type
-- | -- | -- | --
  | StackTrace | "   at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)\r\n   at System.Linq.Enumerable.Select[TSource,TResult](IEnumerable`1 source, Func`2 selector)\r\n   at DotnetCraigslist.PageParser.ParsePosting(PostingRequest request, Stream content)\r\n   at DotnetCraigslist.CraigslistClient.<GetPostingAsync>d__8.MoveNext()\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at DotnetCraigslist.CraigslistStreamingClient.<StreamPostings>d__5.MoveNext()\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at DotnetCraigslist.CraigslistStreamingClient.<StreamPostings>d__5.MoveNext()\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Threading.Tasks.Sources.ManualResetValueTaskSourceCore`1.ThrowForFailedGetResult(Int16 token)\r\n   at System.Threading.Tasks.Sources.ManualResetValueTaskSourceCore`1.GetResult(Int16 token)\r\n   at DotnetCraigslist.CraigslistStreamingClient.<StreamPostings>d__5.System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult(Int16 token)\r\n   at System.Runtime.CompilerServices.ValueTaskAwaiter`1.GetResult()\r\n   at Program.<<Main>$>d__0.MoveNext() in C:\\Projects\\deleteme\\deleteme\\Program.cs:line 13\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at Program.<<Main>$>d__0.MoveNext() in C:\\Projects\\deleteme\\deleteme\\Program.cs:line 13" | string



</body>
</HTML>

It outputs 2 of the results, then excepts on the third, which right now happens to be a link to this entry https://losangeles.craigslist.org/wst/ele/d/los-angeles-amd-matx-gaming-computer/7518591985.html

There's nothing that seems off about this page.

**Comment by wesleythorsen** [comment]
I was able to reproduce, there was a bug with version `0.2.0`. The bug has since been fixed and is deployed in the latest version (`0.4.0`).

**Comment by stoopdapoop** [comment]
awesome, I can't try it till this friday, but I'll close this now and just re-open if I still have issues.

**Comment by stoopdapoop** [comment]
Hey there, so I'm running 0.4.0 and it was working for a couple of days, but now I get a different exception when running the same code as above.

`<HTML>
<head>
<title>Document</title></head>
<body>



  | Name | Value | Type
-- | -- | -- | --
◢ | $exception | {"Value cannot be null. (Parameter 'source')"} | System.ArgumentNullException
  | ▶ Data | {System.Collections.ListDictionaryInternal} | System.Collections.IDictionary {System.Collections.ListDictionaryInternal}
  | HResult | -2147467261 | int
  | HelpLink | null | string
  | ▶ InnerException | null | System.Exception
  | Message | "Value cannot be null. (Parameter 'source')" | string
  | ParamName | "source" | string
  | Source | "System.Linq" | string
  | StackTrace | "   at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)\r\n   at System.Linq.Enumerable.Select[TSource,TResult](IEnumerable`1 source, Func`2 selector)\r\n   at DotnetCraigslist.PageParser.ParsePosting(PostingRequest request, Stream content)\r\n   at DotnetCraigslist.CraigslistClient.<GetPostingAsync>d__8.MoveNext()\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at DotnetCraigslist.CraigslistStreamingClient.<StreamPostings>d__5.MoveNext()\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at DotnetCraigslist.CraigslistStreamingClient.<StreamPostings>d__5.MoveNext()\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Threading.Tasks.Sources.ManualResetValueTaskSourceCore`1.ThrowForFailedGetResult(Int16 token)\r\n   at System.Threading.Tasks.Sources.ManualResetValueTaskSourceCore`1.GetResult(Int16 token)\r\n   at DotnetCraigslist.CraigslistStreamingClient.<StreamPostings>d__5.System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult(Int16 token)\r\n   at System.Runtime.CompilerServices.ValueTaskAwaiter`1.GetResult()\r\n   at Program.<<Main>$>d__0.MoveNext() in C:\\Projects\\deleteme\\deleteme\\Program.cs:line 13\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at Program.<<Main>$>d__0.MoveNext() in C:\\Projects\\deleteme\\deleteme\\Program.cs:line 13" | string
  | ▶ TargetSite | {Void ThrowArgumentNullException(System.Linq.ExceptionArgument)} | System.Reflection.MethodBase {System.Reflection.RuntimeMethodInfo}
  | ▶ Static members |   |  
  | ▶ Non-Public members |   |  



</body>
</HTML>
`

**Comment by stoopdapoop** [comment]
So I was able to dig and find what was causing the problem. Under some circumstances craigslist returns a page that states 

```
"We've detected you are using a browser that is missing critical features.
Please visit craigslist from a modern browser."
```

This page doesn't have any results to parse, so the results parser crashes when trying access them.  I wonder what can be done to prevent this.
